### PR TITLE
Migrating from React.PropTypes

### PR DESCRIPTION
**Pull request by chadams**
using prop-types package for newer React version 15.5.0

### CODE REVIEW

**Comment by chadams** [comment]
I went ahead and added both react updates to same branch as they were small

**Comment by afc163** [comment]
:+1:

**Comment by probablyup** [comment]
@JedWatson without these changes, using the component with React >=15.5 emits red console errors that look like hard errors (thanks Facebook 😞); would be great to get this in soon

**Comment by afc163** [comment]
ping

**Comment by derpoho** [comment]
@JedWatson Can you please merge this? Since the React 16 Update is coming soon, this is seriously necessary!

File: src/Hammer.js
Comment (derpoho) [inline src/Hammer.js]: Setting Statics will not work with Webpack if someone doesn't compile the node_modules folder. For the sake of compatibility you need to add PropTypes and displayName after instantiating the class:

```
HammerComponent.displayName = 'Hammer'
HammerComponent.propTypes = {
	className: PropTypes.string,
}
```

File: src/Hammer.js
Comment (derpoho) [inline src/Hammer.js]: Here:
```
HammerComponent.displayName = 'Hammer'
HammerComponent.propTypes = {
	className: PropTypes.string,
}
```

**Review by derpoho — CHANGES_REQUESTED** [review]

**Comment by derpoho** [comment]
@chadams Maybe you can implement the suggested changes, then we could use this PR.

**Comment by paranoidjk** [comment]
@derpoho can you merge this and continue fix the suggested changes, since we need to fix the react warning.

**Comment by paranoidjk** [comment]
@derpoho or i can open a another PR after you merge this

**Comment by derpoho** [comment]
@paranoidjk I can't merge it, i have no rights in this Repo, you must ask @chadams ;)

**Comment by afc163** [comment]
I don't think this is still maintained, maybe we should maintain a forked version? @paranoidjk

**Comment by paranoidjk** [comment]
@afc163 

Ok, i will fork this repo to [react-component](https://github.com/react-component/) group and continue to mantain it.

still, thanks for @JedWatson great job.

File: src/Hammer.js
Comment (JedWatson) [inline src/Hammer.js]: We're not using babel, so using the new class syntax will break browser support. We need to use the `create-react-class` package instead and leave this as-is

File: package.json
Comment (JedWatson) [inline package.json]: this should be added as a dependency, otherwise projects depending on `react-hammerjs` won't have it installed

**Review by JedWatson — CHANGES_REQUESTED** [review]
Thanks for the PR! would you mind fixing these two things please? then I can merge it and publish a new version

**Comment by JedWatson** [comment]
Just FYI - I've been away for personal reasons but am picking this up at the moment and getting it ready for react 16, plus merging some other fixes.

@afc163 @paranoidjk I don't mind if you want to continue to maintain a fork, but you're welcome to help maintain this one if you like, just let me know.

**Comment by pista329** [comment]
Any progress on this guys?

**Comment by paranoidjk** [comment]
@JedWatson 💯 
Since our project https://ant.design/ is heavily rely on hammerjs, so  I'd like to help with mantain this one.

If you agree with this, please add me owner permission,  both my github and npm id is `paranoidjk`, email is `hust2012jiangkai@gmail.com`, thanks.

> our fork is https://github.com/react-component/react-hammerjs

**Comment by ibc** [comment]
@JedWatson, I do not mean to hurry but any idea when this PR will be merged and a new NPM version will be available?

**Comment by chadams** [comment]
Sorry missed these comments. However, I'm not understanding what needs to be done.

**Comment by pista329** [comment]
Hey guys, this should do the work. https://github.com/JedWatson/react-hammerjs/pull/74

**Comment by chadams** [comment]
I removed the statics, is thats what is holding this up?

File: src/Hammer.js
Comment (chadams) [inline src/Hammer.js]: Done, are there any official tests for hammer, or some good place to test this change?

**Comment by quantumew** [comment]
I can't use this wonderful tool in prod if it takes months to get fixes like these in. Perhaps another owner/maintainer should be added to this project. It is a pity because this tool is so easy.

**Comment by wwqrd** [comment]
@chadams Did you ever find out?

**Comment by jrsinclair** [comment]
If I understand correctly, what's holding this up is that `prop-types` needs to be moved in package.json from `devDependencies` to `dependencies`. Can anyone confirm this?

**Comment by synaestheory** [comment]
@chadams @jrsinclair The move from devDependencies to dependencies is the only comment on the PR where changes were requested.  I would guess that's all that's holding this up.

**Comment by wwqrd** [comment]
Looks like a version with the amends has already been posted here: #74

**Comment by jrsinclair** [comment]
This just became slightly more pressing, now that React 16 has been released.

**Comment by jrsinclair** [comment]
Was this PR made redundant by #78?

**Comment by JedWatson** [comment]
There were a bunch of issues with this in the end, I've cleaned it up significantly and replaced the build system so it should be good now. Try using `1.0.1`

**Pull request closed without merging**
